diff --git a/miniclib.c b/miniclib.c
index 910b1af..7cd928c 100755
--- a/miniclib.c
+++ b/miniclib.c
@@ -4,7 +4,9 @@
 
 #include "miniclib.h"
 
+#if !defined(_MSC_VER) || (_MSC_VER < 1800)
 int errno = ESUCCESS;
+#endif
 
 FILE *stdin=NULL, *stdout=NULL, *stderr=NULL;
 
diff --git a/miniclib.h b/miniclib.h
index 19e51d1..7ad0190 100755
--- a/miniclib.h
+++ b/miniclib.h
@@ -19,10 +19,16 @@ typedef unsigned short wchar_t;
 /* you must call this before using any other function!!! */
 void mCRTinit(void);
 
+#if defined(_MSC_VER) && (_MSC_VER >= 1800)
+#include <stddef.h>
+#include <time.h>
+#include <errno.h>
+#else
 typedef unsigned long time_t;
 typedef unsigned size_t;
 
 extern int errno;
+#endif
 
 #ifndef NULL
 #ifdef  __cplusplus
@@ -170,8 +176,8 @@ FILE *_fdopen(int handle, const char *mode);
 #define WIN32_LEAN_AND_MEAN
 #define _INC_STRING		/* don't allow windows.h to include MS clib's string.h */
 #define _INC_MEMORY
-#define _INC_EXCPT      /* don't allow VC8+SDK2003R2 to include MS clib */
 #if (_MSC_VER >= 1500)
+#if !defined(_EXCEPTION_DISPOSITION)
 typedef enum _EXCEPTION_DISPOSITION {
     ExceptionContinueExecution,
     ExceptionContinueSearch,
@@ -179,6 +185,10 @@ typedef enum _EXCEPTION_DISPOSITION {
     ExceptionCollidedUnwind
 } EXCEPTION_DISPOSITION;
 #endif
+#ifndef _INC_EXCPT
+#define _INC_EXCPT      /* don't allow VC8+SDK2003R2 to include MS clib */
+#endif
+#endif
 #define _INC_CTYPE
 #define __STRALIGN_H_
 #include <windows.h>
